src/Interpreters/SubqueryForSet.cpp

```diff
@@ -39,10 +39,20 @@ void SubqueryForSet::renameColumns(Block & block)
     }
 }
 
-void SubqueryForSet::setJoinActions(ExpressionActionsPtr actions)
+void SubqueryForSet::addJoinActions(ExpressionActionsPtr actions)
 {
     actions->execute(sample_block);
-    joined_block_actions = actions;
+    if (joined_block_actions == nullptr)
+    {
+        joined_block_actions = actions;
+    }
+    else
+    {
+        auto new_dag = ActionsDAG::merge(
+            std::move(*joined_block_actions->getActionsDAG().clone()),
```

Comment:
Should we split `joined_block_actions` into separate variables not to use `merge` here? Now it consist from [createJoinedBlockActions](https://github.com/ClickHouse/ClickHouse/blob/a6ea5ff682a2233f9d7983b4459012ec01ba1cb2/src/Interpreters/ExpressionAnalyzer.cpp#L763) and additional converting step. @KochetovNicolai

Reply:
As I understand, this actions are applied to right part of join. And we don't use separate step for this actions for now. So, I think it is ok to merge.